### DISCUSSION TITLE
Introduce a Module with a Fast Detachable ThreadLocal Implementation

### DISCUSSION THREAD

**Discussion by sirinath** [activej/activej#182] [Ideas]
Netty has a [FastThreadLocal implementation](https://netty.io/4.0/api/io/netty/util/concurrent/FastThreadLocal.html). Can a similar but faster thread local implementation be added to ActiveJ?

The ThreadLocal can have primitive specialisations to avoid boxing.

Following usages can be replaced once introduced: https://github.com/activej/activej/search?q=ThreadLocal

Also ThreadLocalRandom can be replaced: https://github.com/activej/activej/search?q=ThreadLocalRandom

Also see:
- https://developpaper.com/what-the-hell-is-fastthreadlocal-the-existence-of-threadlocal/
- https://github.com/raphw/weak-lock-free/blob/master/src/main/java/com/blogspot/mydailyjava/weaklockfree/DetachedThreadLocal.java

**Comment by eduard-vasinskyi** [comment]
Hello, @sirinath 
Thank you for the suggestion! Right now, due to our multithreaded model based on  DI-assisted workers, there is practically no multithreading overhead in ActiveJ, in particular,  in using `ThreadLocal`. Nevertheless, it is a good idea to add this optimization in future releases of ActiveJ, so please stay tuned!

**Comment by sirinath** [comment]
Will be availing the for own `ThreadLocal`
